src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+                        ? resultLength + (this.has_more ? 0 : 0)
+                        : (this.index -
+                              1 +
+                              resultLength +
+                              (this.has_more ? 1 : 0)) %
+                          (resultLength + (this.has_more ? 1 : 0));
+
+                this.setActive(resultLength);
+                break;
+            case "ArrowDown":
+                this.index =
+                    this.index === -1 ||
+                    (this.has_more && this.index === resultLength)
+                        ? 0
+                        : (this.index + 1) %
+                          (resultLength + (this.has_more ? 1 : 0));
+                this.setActive(resultLength);
+                break;
+
+            case "Enter":
+                if (this.index === resultLength) {
+                    this.loadmoreentry(e);
+                } else {
+                    this.setInputValue();
+                }
+                break;
+        }
+    }
+    InputEvent(e: any) {
+        this.prevIndex = 0;
+        this.index = -1;
+        this.input_el = e.target;
+        if (this.list) {
+            if ((this.input_el as HTMLInputElement)?.value) {
+                this.filtered_list = this.list.filter((item) =>
+                    item[this.listkey].includes(
+                        (this.input_el as HTMLInputElement)?.value
+                    )
+                );
+
+                this.setSuggestion(this.filtered_list, false);
+            } else {
+                this.filtered_list = [];
+
+                this.setSuggestion(this.filtered_list, false);
+            }
+        }
+        return;
+    }
+
+    setActive(resultLength: number) {
+        if (this.prevIndex >= 0 && this.prevIndex < resultLength) {
+            const prevItem = this.private_items[this.prevIndex];
+            if (prevItem) {
+                prevItem.classList.remove("sgn-active");
+            }
+        }
+        if (this.index >= 0 && this.index < resultLength) {
+            const selectedItem = this.private_items[this.index];
+
+            if (selectedItem) {
+                if (this.prevIndex != -1) {
+                    this.private_items[this.prevIndex].classList.remove(
+                        "sgn-active"
+                    );
+                }
+                selectedItem.classList.add("sgn-active");
+            }
+        } else if (this.index == resultLength && this.has_more) {
+            this.private_items[this.index].classList.add("sgn-active");
+        }
+
+        this.prevIndex = this.index;
+    }
+
+    setInputValue() {
+        const selectedValue = this.result[this.index]?.name;
+        this.prevIndex = this.index;
+        if (selectedValue) {
+            (this.input_el as HTMLInputElement).value = selectedValue;
+        }
+    }
+    _highlight(name: any) {
+        const input = (this.input_el as HTMLInputElement)?.value;
```

Comment:
_highlight() is called in loop, and we are calling lowerCase() of input value each time. It will be better to do it once and passed as  a parameter to this function so that we can avoid calling toLowerCase in loop